src/App.tsx

```diff
@@ -1,61 +1,153 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './components/types/User';
+import { TodoWithUser } from './components/types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+
+const preparedTodos: TodoWithUser[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(({ id }) => id === todo.userId) || null,
+} as TodoWithUser));
+
+function findUserById(userId: number): User {
+  return usersFromServer.find(({ id }) => id === userId) as User;
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState<TodoWithUser[]>(preparedTodos);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userIdError, setUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const handleBtAdd = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    let isError = false;
+
+    if (title === '') {
```

Comment:
```suggestion
    if (!title) {
```
Don't compare with an empty string, just check if the value exists.
[Falsy values article](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)

The same for the userId.